### ISSUE TITLE
Show amount converted in FIAT via CoinGecko API

### ISSUE THREAD

**Issue by tiero** [vulpemventures/dex.vulpem.com#5]
The User should set the FIAT currency of reference or use default ie. USD (maybe navbar top-right option?)

In the conversion form, when calculating the price preview, we should also convert those amounts to the equivalent in the reference FIAT asset.

At the moment we support only L-BTC, USDT and LCAD, on Coingeko we can only retrieve price of BTC (ie. 1 LBTC = 1 BTC) and USDT (which is different than USD, it fluctuates a bit). For LCAD we should not show it, since not listed on CoinGecko. 

Reference implementation: https://github.com/tdex-network/tdex-app

**Comment by bordalix** [comment]
First of all, a couple of notes:

1. Nav top right is where the Marina action button is, maybe it is not a good idea to add more controls there;
2. We just need to show the fiat value once, for the trade, and not for each token;
3. Since the value in fiat is the same for both tokens of the pair, we don’t have an issue with the fact of L-CAD not being available on the Coingecko API: the other cryptocurrency on the pair will indicate the fiat value.

### What to show?

Checking the [API documentation](https://www.coingecko.com/en/api/documentation) we see support for a [lot of different currencies](https://api.coingecko.com/api/v3/simple/supported_vs_currencies). See image [1]

Supported currencies:
- Should we support all of them?
- Is there any audience analysis with most visited countries?
- Or should we go for only USD and EUR?

Another option is to use Marina’s selected currency, which only supports USD, EUR and GBP. In this case, the default currency for the site would be the same as Marina’s, and currency changes would only occur inside the Marina’s UI. I’m discarding this option for now since it would probably create a development dependency between the two different projects.

### How to show?

Don’t use symbols ($, €), use the ticker instead (USD, EUR)

When showing the fiat value to the user, use `toLocaleString()` to have all values shown with the user’s expected format (comma vs dot decimal separator, thousands separator, …)

### When to show?

The fiat value for the trade should be calculated (and shown) at the same time as the value for the pair token is calculated and shown.

### Where to show?

Currency selection format:
- If we want to support all currencies supported by Coingecko, then we should go for a modal solution;
- If not, a simple dropdown would suffice.

Possible locations for the fiat value:
- At the same level of the h1 (“Trade”), but right aligned and with smaller font. This is my preferred option, since the value is related with the trade, not the tokens per si. See mockup [2];
- Somewhere between the CoinRows and the buttons.

Possible locations for the currency selector:
- Use a currency selector at the nav top left that will open the modal;
- Make a link from the fiat value that will open the modal. This is my preferred option, will eliminate clutter from the UI, and with a good default currency, users will not see the need to use it. See mockup [2].

Default currency:
- Unfortunately there’s no browser support for currency;
- Use localStorage to store user’s chosen currency;
- If no chosen currency, use USD.

Fetching data:
- In order to spare API calls to Coingecko, the site should, on startup, request the fiat value for L-BTC and Tether for the default currency, and store it in the session;
- Every change on the input fields will use this stored value to calculate the fiat value;
- New requests to the API are to be made only when the selected currency changes;
- There are no CORS limitations on the Coingecko API, so the site can fetch the information directly.

Note, we could implement a provider on Marina to fetch this information for us, but again, it could create a development dependency between the two projects.

### Images
1. [List of supported currencies](https://i.ibb.co/0DRVrWJ/supported-currencies.png)
2. [Mockup showing fiat value](https://i.ibb.co/SN9TmRZ/show-fiat.png)

### TL/DR
1. Use only USD + EUR
2. Show fiat value as show in mockup
3. Clicking fiat value will show dropdown with USD + EUR options
4. Fetch directly from the Coingecko API

**Comment by tiero** [comment]
> Since the value in fiat is the same for both tokens of the pair, we don’t have an issue with the fact of L-CAD not being available on the Coingecko API: the other cryptocurrency on the pair will indicate the fiat value.

Agreed. we just skip the asset which are not listed on CoinGecko

> ## Where to show?

<img width="508" alt="image" src="https://user-images.githubusercontent.com/3596602/143898526-f9cf9adf-42be-43a7-b7f3-f864e1fc3b63.png">

I think this reference from Uniswap is a nice base and I would suggest to do in a very similar way. 

> If we want to support all currencies supported by Coingecko, then we should go for a modal solution;

Would use this solution, more future proof. Still I will just support a subset of FIAT currencies: `USD`, `EUR` and `CAD` with USD default.

> Make a link from the fiat value that will open the modal

Yes I agree

> it could create a development dependency between the two projects

agreed. Marina should not be aware and let the App developers to implement these UX details re: fiat conversion
